### ISSUE TITLE
Unsupervised VOS

### ISSUE THREAD

**Issue by dipikaboro2** [Hydragon516/GSANet#3]
Hello. Congratulations on the great work.
I am new to unsupervised learning in Computer Vision.
My question is -
    1. In Unsupervised VOS is it OK to compute loss using ground truth during training or pre-training?
    2. In your previous answer you mentioned that supervised or unsupervised depends on the whether mask is given for the first frame for test data. However, in the codebase, I found that for all the RGB images, the corresponding masks (GT) are also provided during testing. [ https://github.com/Hydragon516/GSANet/blob/1a70dada4bb2599db6b2f60eadba496378280c9c/train.py#L111 ]

**Comment by Hydragon516** [comment]
Thank you for your interest of our works.

1. Regarding the use of ground truth during training or pre-training in Unsupervised VOS:
In the context of Unsupervised Video Object Segmentation (VOS), the term "unsupervised" means that there is no explicit information about the target object (such as class names or target masks) provided during the training or pre-training phase. This is consistent with the definition used in existing unsupervised VOS literature. Therefore, it is not appropriate to compute the loss using ground truth masks during training or pre-training for an unsupervised VOS task.

2. Regarding the use of ground truth masks during testing:
The ground truth (GT) masks provided during testing are solely for performance evaluation purposes. They are not used as inputs to the model. 

[https://github.com/Hydragon516/GSANet/blob/1a70dada4bb2599db6b2f60eadba496378280c9c/train.py#L145](https://github.com/Hydragon516/GSANet/blob/1a70dada4bb2599db6b2f60eadba496378280c9c/train.py#L145)

**Comment by dipikaboro2** [comment]
Thank you for your response.

I found that during training phase the loss is computed using the ground truths. This is in train function of train.py

 https://github.com/Hydragon516/GSANet/blob/1a70dada4bb2599db6b2f60eadba496378280c9c/train.py#L94

**Comment by Hydragon516** [comment]
I think you misunderstand the UVOS task. I think you should refer to other papers in this [link](https://github.com/suhwan-cho/awesome-video-object-segmentation?tab=readme-ov-file#unsupervised-vos-papers).
